### PR TITLE
Use a multimodule maven organisation for 2.x

### PR DESCRIPTION
**Pull request by victornoel**
- Change groupId to org.pac4j.jax-rs
 - Users should depend on jersey-pac4j or resteasy-pac4j
 - Users can still depend on org.pac4j.jax-rs-pac4j for backward compatibility
 - Add saveInSession parameter for callback endpoints

cf #37 for details on why

@pierre-l here it is for 2.x

@leleuj, two points:
 - the difference with #37 is that there is an extra project `org.pac4j.jax-rs-pac4j` for backward compatibility
 - I understood from the release note said that pac4j 2.3.0 had the new `saveInSession` parameter for callback endpoints but it isn't there in `CallbackLogic.perform`, instead I set it on the logic itself, could you confirm that what I did is correct?

### CODE REVIEW

**Comment by leleuj** [comment]
OK. This one targets an existing branch, so yes, it makes sense to keep the existing `jax-rs-pac4j` library.

Yes, pac4j v2.3.0 has a `saveSession` property in the `DefaultCallbackLogic` and not in the `perform` method to keep backward compatiblity.

File: core/src/main/java/org/pac4j/jax/rs/annotations/Pac4JCallback.java [hunk 1]
```diff
@@ -57,4 +57,13 @@
...
     boolean[] multiProfile() default {};
+
+    /**
+     * Note that this parameter only takes one value at most: empty array (default) is used to represent default pac4j
+     * setting, one boolean will be used by the filter, and more than one boolean will fail the resource method
+     * initialisation.
+     * 
+     * @return value for {@link CallbackFilter#setSaveInSession(Boolean)}
+     */
+    boolean[] saveInSession() default {};
```
Comment (leleuj) [inline core/src/main/java/org/pac4j/jax/rs/annotations/Pac4JCallback.java]: Why not using `Boolean`?

Comment (victornoel) [reply core/src/main/java/org/pac4j/jax/rs/annotations/Pac4JCallback.java]: Why would you want something that can have the value `null` here?

The point of those annotations is that they allow the user to express themselves without over thinking: if they put a value it is used, if they do not, the default is used.

Comment (leleuj) [reply core/src/main/java/org/pac4j/jax/rs/annotations/Pac4JCallback.java]: OK

**Comment by victornoel** [comment]
I updated the PR, I will merge it later tonight except if you have other feedbacks :)

**Pull request merged**